app/lib/task/cloudcompute/fakecloudcompute.dart

```diff
@@ -104,6 +126,80 @@ class FakeCloudCompute extends CloudCompute {
     _instances.removeWhere((i) => i.instanceName == instanceName);
     _instances.add(instance._copyWith(state: InstanceState.terminated));
   }
+
+  var _running = false;
+  Completer<void> _doneRunningInstance = Completer()..complete();
+
+  /// Start execution of instances.
+  void startInstanceExecution() {
+    if (_running) {
+      throw StateError(
+        'FakeCloudCompute.startInstanceExecution() have already been called!',
+      );
+    }
+    _running = true;
+    scheduleMicrotask(() async => await _keepExecutionLoopAlive());
+  }
+
+  Future<void> _keepExecutionLoopAlive() async {
+    final instance = _instances.firstWhereOrNull(
+      (i) => i.state == InstanceState.pending,
+    );
+    // If not running, there are no pending instance, or an instance is already
+    // running, then we're done.
+    if (!_running || instance == null || !_doneRunningInstance.isCompleted) {
+      return;
+    }
+
+    fakeStartInstance(instance.instanceName);
+    _doneRunningInstance = Completer();
+
+    scheduleMicrotask(() async {
+      _log.info('Starting to run ${instance.instanceName}');
+      try {
+        final proc = await Process.start(
+          Platform.resolvedExecutable,
+          ['run', 'pub_worker', ...instance.arguments],
+          workingDirectory: p.join(resolveAppDir(), '..', 'pkg', 'pub_worker'),
+          mode: ProcessStartMode.inheritStdio,
+        );
+        final exitCode = await proc.exitCode;
+        _log.info('pub_worker exit code: $exitCode');
+      } finally {
+        // Don't terminate the instance if it's already deleted.
+        if (_instances.any((i) => i.instanceName == instance.instanceName)) {
+          fakeTerminateInstance(instance.instanceName);
+        }
+        // Must happen before we start the next instance.
```

Comment:
Nit: can we do any assert here?

Reply:
I'm not sure how, we start the next iteration in the very next line. And the next iteration will refuse to run if it's not completed.